### PR TITLE
Bump golangci-lint to v1.48.0

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
This incorporates new linters as well as the configuration update. New lint reports have been fixed as well.

#### Which issue(s) this PR fixes:

None
#### Does this PR have test?
None

#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Review by jhrozek — APPROVED** [review]
/lgtm

**Review by pjbgf — APPROVED** [review]
/lgtm

**Review by JAORMX — APPROVED** [review]

**Comment by saschagrunert** [comment]
Rebased and fixed a bunch of other lint reports.

File: internal/pkg/daemon/bpfrecorder/generate/main.go [hunk 1]
```diff
@@ -189,16 +189,17 @@ func generateBtf(builder *strings.Builder) error {
 			fmt.Fprintf(docs, "- %s\n", os)
 		}
 		if _, ok := btfs[os][osVersion]; !ok {
+			const indent = 2
 			btfs[os][osVersion] = map[types.Arch]map[types.Kernel][]byte{}
-			fmt.Fprintf(docs, "%s- %s\n", strings.Repeat(" ", 2), osVersion) // nolint:gomnd // indent
+			fmt.Fprintf(docs, "%s- %s\n", strings.Repeat(" ", indent), osVersion) //olint:gomnd // indent
```
Comment (jhrozek) [inline internal/pkg/daemon/bpfrecorder/generate/main.go:194]: `olint` sounds like a typo?

**Review by jhrozek — COMMENTED** [review]
looks good except for the one typo

File: internal/pkg/daemon/bpfrecorder/generate/main.go (hunk 1 above)
Comment (saschagrunert) [reply internal/pkg/daemon/bpfrecorder/generate/main.go]: Yeah this one can be removed completely. :+1:

**Review by jhrozek — APPROVED** [review]
/lgtm

**Pull request merged**